website/src/blogs/bare-metal-to-production-stateful-workloads-openebs-and-rancher-2-x.md

```diff
@@ -0,0 +1,92 @@
+
+This blog will outline the detailed steps we took to achieve our goal of setting up a bare-metal Kubernetes cluster at Hetzner from scratch using Rancher 2.x, Hetzner node driver, OpenEBS, Cert-Manager, Nginx Ingress loadbalancing and the deployment of the Tutem — Hello World app to demonstrate loadbalancing and Let’s Encrypt.
+
+**NOTE:** All nodes are created with Hetzner CX21 type servers. Our root domain is `cooby.tech` and an `A` record called `rancher.cooby.tech` points to the instance IP.
+
+### Initial Setup
+
+We’ll be creating 4 Hetzner cluster servers:
+
+1. `front-1` nginx-ingress loadbalancing (type: worker)
+2. `control-1` control plane (type: etcd, control)
+3. `worker-1` first worker (type: worker)
+4. `worker-2` second worker (type: worker)
+
+- Spin up a Hetzner CX21 instance with a Ubuntu 18.04 image
+- Generate a keypair and make sure you can ssh into the instance
+- Install docker and docker-compose from your local machine:
+
+    curl https://gist.githubusercontent.com/gridworkz/d78c290c4e6fd7753dc21bb50601745a/raw/625fc3b7ddb7a654fe379f67d0bb9c6ac5b9413a/InstallDockerCompose | ssh -i ~/.ssh/<privkey> root@rancher.cooby.tech
+    “/bin/bash -s”
+
+- Login to the instance and add a docker-compose.yml file to run Rancher:
+
+![Login to instance](/public/images/blog/login-to-instance.png)
+- Run: 
+
+    docker-compose up -d
+
+- `Install Hetzner node-driver from:` [https://mxschmitt.github.io](https://mxschmitt.github.io/ui-driver-hetzner/)
+- `Create a node-template for frontend`
+- `Add label loadbalancer=true`
+- `Remove iscsi driver with cloud-init:`
+
+![Remove iscsi driver](/public/images/blog/remove-iscsi-driver.png)
+- Create a node template for workers using the same cloud-config
+
+## Application Configuration
+
+Schedule pods so that they are not running on a node with tag **loadbalancer**. This is not a hard requirement, but in our example cluster it really makes sense.
+![application config](/public/images/blog/application-config.png)
+## Adding Nginx-Ingress
+
+DNS scheme is configured like this:
+
+- `.cooby.tech` A <`loadbalancer machines`>
+
+Which enables you to use:
+
+- **Example 1.** `cooby.tech` and to use nginx ingress to route it to the example1 workload. Also in the DNS server, you can do, for example, cooby.in.montel.care CNAME montel.cooby.tech and again use our nginx-loadbalancer to route cooby.in.montel.care to your desired workload.
+
+**Pro Tip:** Once an Ingress service is created it defaults to port 42. Remember to edit the services’ YAML and change it to the port your application requires. You will save yourself hours of confusion.
+
+## Let’s Encrypt Issuer
+
+The simplest way to add Let’s Encrypt in Rancher is to install it via catalog apps.
+
+After that every ingress that contains the following annotations:
+![ingress](/public/images/blog/ingress.png)
```

Comment:
We can have one line space on the top of the image as well as the bottom of the image. please do same for everywhere.

Reply:
This is for better formatting